v2/tools/generator/internal/codegen/pipeline/json_serialization_test_cases.go

```diff
@@ -104,5 +103,13 @@ func (s *objectSerializationTestCaseFactory) AddTestTo(def astmodel.TypeDefiniti
 	isOneOf := astmodel.OneOfFlag.IsOn(def.Type()) // this is ugly but can’t do much better right now
 
 	testcase := testcases.NewJSONSerializationTestCase(def.Name(), container, isOneOf, s.idFactory)
+	if _, isResource := astmodel.AsResourceType(def.Type()); isResource {
+		// Don't need to test resources many times, the spec and status types are tested independently
+		testcase.SetMinSuccessfulTests(20)
+	} else if astmodel.IsSpec(def.Name()) || astmodel.IsStatus(def.Name()) {
+		// Reduce count of Spec and Status tests to reflect those done by the resource tests
```

Comment:
Maybe worth adding here about the comparison of `Reduce`. Maybe something to tell about the default being 100 and we're reducing to 80

Reply:
I'll add some constants and comments.